### ISSUE TITLE
Spy does not retry when used as an alias

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#1156]
Bug, Mac, Cypress 1.4.1

Reproducible test https://github.com/cypress-io/cypress-example-recipes/issues/86 and pull request https://github.com/cypress-io/cypress-example-recipes/pull/87

```js
// works
it('should retries until spy is called', () => {
    cy.wrap(spy).should('have.been.calledOnce')
    setTimeout(spy, 1000)
})
// does not retry
it('works as alias', () => {
    cy.get('@spy2').should('have.been.calledOnce')
    setTimeout(spy2, 1000)
})
```

**Comment by brian-mann** [comment]
Investigated and this is the root cause for this issue as well: https://github.com/cypress-io/cypress-example-recipes/issues/88

**Comment by brian-mann** [comment]
Released in [`2.0.0`](https://docs.cypress.io/guides/references/changelog.html#2-0-0).
